src/libraries/Microsoft.PowerFx.Core/Public/Config/ComposedReadOnlySymbolTable.cs

```diff
@@ -82,6 +83,21 @@ TexlFunctionSet INameResolver.Functions
             }
         }
 
+        // Expose the list to aide in intellisense suggestions and create new composable type symbol tables.
+        IEnumerable<KeyValuePair<DName, FormulaType>> INameResolver.DefinedTypes
+        {
+            get
+            {
+                foreach (var table in _symbolTables)
+                {
+                    foreach (var type in table.DefinedTypes)
```

Comment:
>  foreach (var type in table.DefinedTypes) [](http://example.com/codeflow?start=18&length=42)

Subtle issue -  is there a shadowing issue where it could show up twice? (see GlobalSymbols for example) 

ST1 defines  Foo=Number
ST2 defines Foo=Text; Bar=Bool

ST3 = Compose(ST1, ST2)

ST3's list should be Foo=Number,Bar=Bool
(Foo only shows up once)

Reply:
sure . will address that